napari/layers/vectors/_slice.py

```diff
@@ -81,21 +83,7 @@ def __call__(self) -> _VectorSliceResponse:
                 request_id=self.id,
             )
 
-        # We want a numpy array so we can use fancy indexing with the non-displayed
```

Comment:
Do you have sense or evidence about performance changes here when not doing thick-slicing? Could we do a before/after run on one of the existing ASV benchmarks (e.g. `benchmark_vectors_layer.Vectors3DSuite.time_set_view_slice`).

Similar question for the other potentially affected layers here - I think that's images and points?

Reply:
I tried a few menchmarks, and it seems that we have a regression, reproducible over a couple runs. TLDR: The regression seems bigger for small data, which is at least promising.

Here's vectors:

```
[ 50.00%] · For napari commit e6815274 <feature/thick-slice-layers> (round 2/2):
[ 50.00%] ·· Benchmarking virtualenv-py3.11
[ 75.00%] ··· benchmark_vectors_layer.Vectors3DSuite.time_set_view_slice                                                                                                                                                              ok
[ 75.00%] ··· ======== =============
               param1
              -------- -------------
                 16     1.28±0.02ms
                 64     1.27±0.02ms
                256     1.35±0.02ms
                1024    1.50±0.01ms
                4096    2.06±0.08ms
               16384    4.16±0.06ms
               65536     12.8±0.2ms
              ======== =============

[ 75.00%] · For napari commit 21bc6244 <main> (round 2/2):
[ 75.00%] ·· Building for virtualenv-py3.11.........................
[ 75.00%] ·· Benchmarking virtualenv-py3.11
[100.00%] ··· benchmark_vectors_layer.Vectors3DSuite.time_set_view_slice                                                                                                                                                              ok
[100.00%] ··· ======== =============
               param1
              -------- -------------
                 16       993±30μs
                 64     1.00±0.03ms
                256     1.09±0.05ms
                1024    1.29±0.06ms
                4096    1.83±0.09ms
               16384    3.83±0.08ms
               65536     12.2±0.2ms
              ======== =============

       before           after         ratio
     [21bc6244]       [e6815274]
     <main>           <feature/thick-slice-layers>
+        993±30μs      1.28±0.02ms     1.29  benchmark_vectors_layer.Vectors3DSuite.time_set_view_slice(16)
+     1.00±0.03ms      1.27±0.02ms     1.27  benchmark_vectors_layer.Vectors3DSuite.time_set_view_slice(64)
+     1.09±0.05ms      1.35±0.02ms     1.23  benchmark_vectors_layer.Vectors3DSuite.time_set_view_slice(256)
+     1.83±0.09ms      2.06±0.08ms     1.12  benchmark_vectors_layer.Vectors3DSuite.time_set_view_slice(4096)

SOME BENCHMARKS HAVE CHANGED SIGNIFICANTLY.
PERFORMANCE DECREASED.
```

And here's points:

```
[ 50.00%] · For napari commit e6815274 <feature/thick-slice-layers> (round 2/2):
[ 50.00%] ·· Benchmarking virtualenv-py3.11
[ 75.00%] ··· benchmark_points_layer.Points3DSuite.time_set_view_slice                                            ok
[ 75.00%] ··· ======== =============
               param1
              -------- -------------
                 16     1.35±0.03ms
                 64     1.40±0.05ms
                256     1.42±0.04ms
                1024    1.52±0.05ms
                4096    1.56±0.04ms
               16384     1.88±0.2ms
               65536    3.25±0.05ms
              ======== =============

[ 75.00%] · For napari commit 21bc6244 <main> (round 2/2):
[ 75.00%] ·· Building for virtualenv-py3.11.........................
[ 75.00%] ·· Benchmarking virtualenv-py3.11
[100.00%] ··· benchmark_points_layer.Points3DSuite.time_set_view_slice                                            ok
[100.00%] ··· ======== =============
               param1
              -------- -------------
                 16     1.09±0.02ms
                 64     1.09±0.02ms
                256     1.15±0.07ms
                1024    1.13±0.01ms
                4096    1.22±0.03ms
               16384    1.57±0.05ms
               65536     3.03±0.4ms
              ======== =============

       before           after         ratio
     [21bc6244]       [e6815274]
     <main>           <feature/thick-slice-layers>
+     1.13±0.01ms      1.52±0.05ms     1.35  benchmark_points_layer.Points3DSuite.time_set_view_slice(1024)
+     1.09±0.02ms      1.40±0.05ms     1.29  benchmark_points_layer.Points3DSuite.time_set_view_slice(64)
+     1.22±0.03ms      1.56±0.04ms     1.28  benchmark_points_layer.Points3DSuite.time_set_view_slice(4096)
+     1.15±0.07ms      1.42±0.04ms     1.23  benchmark_points_layer.Points3DSuite.time_set_view_slice(256)
+     1.09±0.02ms      1.35±0.03ms     1.23  benchmark_points_layer.Points3DSuite.time_set_view_slice(16)
+     1.57±0.05ms       1.88±0.2ms     1.20  benchmark_points_layer.Points3DSuite.time_set_view_slice(16384)

SOME BENCHMARKS HAVE CHANGED SIGNIFICANTLY.
PERFORMANCE DECREASED.
```

Reply:
Also similar to the above (failed ones are because my laptop doens't have enough memory):

```
[ 50.00%] ··· Running (benchmark_image_layer.Image3DSuite.time_set_view_slice--).
[ 50.00%] · For napari commit e6815274 <feature/thick-slice-layers~1> (round 2/2):
[ 50.00%] ·· Benchmarking virtualenv-py3.11
[ 75.00%] ··· benchmark_image_layer.Image3DSuite.time_set_view_slice                                                                                                                                                          1/7 failed
[ 75.00%] ··· ======== =============
               param1
              -------- -------------
                 16     1.13±0.02ms
                 32     1.13±0.01ms
                 64     1.14±0.01ms
                128     1.13±0.03ms
                256     1.16±0.03ms
                512     1.20±0.04ms
                1024       failed
              ======== =============

[ 75.00%] · For napari commit b5ecd293 <main> (round 2/2):
[ 75.00%] ·· Building for virtualenv-py3.11..............
[ 75.00%] ·· Benchmarking virtualenv-py3.11
[100.00%] ··· benchmark_image_layer.Image3DSuite.time_set_view_slice                                                                                                                                                          1/7 failed
[100.00%] ··· ======== =============
               param1
              -------- -------------
                 16       969±7μs
                 32       960±9μs
                 64       966±10μs
                128       975±10μs
                256     1.03±0.01ms
                512     1.02±0.02ms
                1024       failed
              ======== =============

       before           after         ratio
     [b5ecd293]       [e6815274]
     <main>           <feature/thick-slice-layers~1>
+        966±10μs      1.14±0.01ms     1.18  benchmark_image_layer.Image3DSuite.time_set_view_slice(64)
+         960±9μs      1.13±0.01ms     1.18  benchmark_image_layer.Image3DSuite.time_set_view_slice(32)
+     1.02±0.02ms      1.20±0.04ms     1.18  benchmark_image_layer.Image3DSuite.time_set_view_slice(512)
+         969±7μs      1.13±0.02ms     1.17  benchmark_image_layer.Image3DSuite.time_set_view_slice(16)
+        975±10μs      1.13±0.03ms     1.16  benchmark_image_layer.Image3DSuite.time_set_view_slice(128)
+     1.03±0.01ms      1.16±0.03ms     1.13  benchmark_image_layer.Image3DSuite.time_set_view_slice(256)

SOME BENCHMARKS HAVE CHANGED SIGNIFICANTLY.
PERFORMANCE DECREASED.
```